modules/social_features/social_group/modules/social_group_request/social_group_request.tokens.inc

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * @file
+ * Builds placeholder replacement tokens for Social Group Request module.
+ */
+
+use Drupal\Core\Render\BubbleableMetadata;
+use Drupal\Core\Url;
+
+/**
+ * Implements hook_token_info().
+ */
+function social_group_request_token_info() {
+
+  $type = [
+    'name' => t('Social Group Request'),
+    'description' => t('Tokens from the social group request module.'),
+  ];
+
+  $social_group_request['request_page_link'] = [
+    'name' => t('Request page link.'),
+    'description' => t('The link to the request page of the group.'),
+  ];
+
+  return [
+    'types' => ['social_group_request' => $type],
+    'tokens' => [
+      'social_group_request' => $social_group_request,
+    ],
+  ];
+}
+
+/**
+ * Implements hook_tokens().
+ */
+function social_group_request_tokens($type, $tokens, array $data, array $options, BubbleableMetadata $bubbleable_metadata) {
+  $replacements = [];
+
+  if ($type === 'social_group_request' && !empty($data['message'])) {
+    /** @var \Drupal\message\MessageInterface $message */
+    $message = $data['message'];
+
+    foreach ($tokens as $name => $original) {
+      switch ($name) {
```

Comment:
We tend to create switch cases in foreach loops for one single case. That doesn't make a lot of sense.

Let's change this so a normal `if` statement.

Reply:
Changed